code/modules/cm_aliens/structures/special/egg_morpher.dm

```diff
@@ -127,20 +162,54 @@
 
 	HasProximity(target)
 
-/obj/effect/alien/resin/special/eggmorph/HasProximity(atom/movable/AM as mob|obj)
+/obj/effect/alien/resin/special/eggmorph/Crossed(atom/movable/AM)
+	HasProximity(AM)
+
+//The invisible traps around the morpher to tell it there's a mob right next to it.
+/obj/effect/egg_trigger
+	name = "egg trigger"
+	icon = 'icons/effects/effects.dmi'
+	anchored = TRUE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	invisibility = INVISIBILITY_MAXIMUM
+
+/obj/effect/egg_trigger/New(loc, obj/effect/alien/egg/source_egg, obj/effect/alien/resin/special/eggmorph/source_eggmorph)
+	..()
+	linked_egg = source_egg
+	linked_eggmorph = source_eggmorph
+
+
+/obj/effect/egg_trigger/Crossed(atom/movable/AM)
+	if(!linked_egg && !linked_eggmorph) //something went very wrong.
+		qdel(src)
+	else if(linked_egg && (get_dist(src, linked_egg) != 1 || !isturf(linked_egg.loc))) //something went wrong
+		forceMove(linked_egg)
+	else if(linked_eggmorph && (get_dist(src, linked_eggmorph) != 1 || !isturf(linked_eggmorph.loc))) //something went wrong
+		forceMove(linked_eggmorph)
+	else if(iscarbon(AM))
+		var/mob/living/carbon/Carbon = AM
+		if(linked_egg && can_hug(AM))
+			linked_egg.HasProximity(Carbon)
+		if(linked_eggmorph && can_hug(AM))
+			linked_eggmorph.HasProximity(Carbon)
```

Comment:
just make the conditions return to avoid the elseifs

ie

```
	if(!linked_egg && !linked_eggmorph) //something went very wrong.
		qdel(src)
		return

	if(linked_egg && (get_dist(src, linked_egg) != 1 || !isturf(linked_egg.loc))) //something went wrong
		forceMove(linked_egg)
		return

	if(linked_eggmorph && (get_dist(src, linked_eggmorph) != 1 || !isturf(linked_eggmorph.loc))) //something went wrong
		forceMove(linked_eggmorph)

		etc
```